projects/storefrontlib/src/lib/store-finder/components/store-finder-search-result/store-finder-search-result.component.ts

```diff
@@ -61,15 +56,17 @@ export class StoreFinderSearchResultComponent implements OnInit, OnDestroy {
       this.searchQuery.useMyLocation
     );
 
-    this.isLoading$ = this.store.pipe(select(fromStore.getStoresLoading));
-    this.locations$ = this.store.pipe(select(fromStore.getFindStoresEntities));
+    this.isLoading$ = this.storeFinderService.getStoresLoading();
+    this.locations$ = this.storeFinderService.getFindStoresEntities();
     this.ngUnsubscribe = this.locations$
       .pipe(map(data => data.geolocation))
       .subscribe(geoData => (this.geolocation = geoData));
```

Comment:
can use `tap` to replace this `subscribe`

Reply:
You mean something like this?

```
this.locations$ = this.storeFinderService.getFindStoresEntities().pipe(
    tap(data => (this.geolocation = data.geolocation))
);
```

Reply:
We can remove the function `private initialize(params: Params)`.
In `ngOnInit()`, we can do this:
```
ngOnInit() {
    this.route.queryParams.subscribe(params => {
      this.searchQuery = this.parseParameters(params);
      this.storeFinderService.findStores(
        this.searchQuery.queryText,
        this.searchQuery.useMyLocation
      );
    });

    this.isLoading$ = ....;
    this.locations$ = this.storeFinderService.getFindStoresEntities().pipe(
    tap(data => (this.geolocation = data.geolocation))
);
      
  }
```

Reply:
btw, remove `void` from `ngOnInit` and `ngOnDestroy`.

Reply:
Refactor done, also i removed `void`.